### ISSUE TITLE
System.Security.Authentication.AuthenticationException occurred once while sending mail 

### ISSUE THREAD

**Issue by deepakgupta03** [sendgrid/sendgrid-csharp#1049]
Hi, 

We got the below exception once while sending a mail using send grid API.
Exception: System.Security.Authentication.AuthenticationException
Message: An error occurred while sending the request. The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel. The remote certificate is invalid according to the validation procedure. 
StackTrace: 
```
System.Net.Http.HttpRequestException:
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at SendGrid.Web+<DeliverAsync>d__0.MoveNext (SendGridMail, Version=6.1.0.0, Culture=neutral, PublicKeyToken=2ae73662c35d80e4)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
```
Thanks in advance.

**Comment by eshanholtz** [comment]
Hi @deepakgupta03 

Can you please [log the request body](https://github.com/sendgrid/sendgrid-csharp/blob/main/TROUBLESHOOTING.md#viewing-the-request-body) as well as the [full error output](https://github.com/sendgrid/sendgrid-csharp/blob/main/TROUBLESHOOTING.md#error-messages) and post the contents of both in a follow up comment so we can further help you debug? Also, what versions of the sendgrid-csharp library and .NET are you using?

**Comment by deepakgupta03** [comment]
Hi @eshanholtz,

Thanks for your response.

We are using the send grid 6.1.0 library and .Net version is 4.7.1.

I was not able to capture the request body. Below is the  error output details that I was able to capture:
**Event time**: 9/14/2020, 20:54
**Message**: An error occurred while sending the request. The underlying connection was closed: Could not establish trust relationship for the SSL/TLS secure channel. The remote certificate is invalid according to the validation procedure. 
**Stack trace:** 
System.Net.Http.HttpRequestException:
Inner exception System.Net.WebException handled at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess:
   at System.Net.HttpWebRequest.EndGetRequestStream (System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Net.Http.HttpClientHandler.GetRequestStreamCallback (System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)
Inner exception System.Security.Authentication.AuthenticationException handled at System.Net.HttpWebRequest.EndGetRequestStream:
   at System.Net.TlsStream.EndWrite (System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Net.PooledStream.EndWrite (System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Net.ConnectStream.WriteHeadersCallback (System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
